icu4c/source/i18n/dtitvfmt.cpp

```diff
@@ -863,6 +863,7 @@ DateIntervalFormat::initializePattern(UErrorCode& status) {
                 setPatternInfo(UCAL_DATE, nullptr, &pattern, fInfo->getDefaultOrder());
                 setPatternInfo(UCAL_MONTH, nullptr, &pattern, fInfo->getDefaultOrder());
                 setPatternInfo(UCAL_YEAR, nullptr, &pattern, fInfo->getDefaultOrder());
+                setPatternInfo(UCAL_ERA, nullptr, &pattern, fInfo->getDefaultOrder());
```

Comment:
The pattern you are adding here does not include the era. You should probably call getBestPattern() after prepending "G" to the timeSkeleton (similar to what you are doing in line 894).